testcase/testcase.go

```diff
@@ -227,22 +213,41 @@ func NewFromFile(path string) (*TestCaseSet, error) {
 // file and the two files are passed to "diff -u". If quiet is true,
 // the progress messages normally written to stderr will be emitted
 // and the output of the diff program will be discarded.
-func (tcs *TestCaseSet) Compare(events []logstash.Event, quiet bool, diffCommand []string) error {
-	result := ComparisonError{
-		ActualCount:   len(events),
-		ExpectedCount: len(tcs.ExpectedEvents),
-		Mismatches:    []MismatchedEvent{},
-	}
+func (tcs *TestCaseSet) Compare(events []logstash.Event, diffCommand []string, liveProducer observer.Property) (bool, error) {
+	status := true
 
 	// Don't even attempt to do a deep comparison of the event
 	// lists unless their lengths are equal.
-	if result.ActualCount != result.ExpectedCount {
-		return result
+	if len(tcs.ExpectedEvents) != len(events) {
+		comparisonResult := &logstash_observer.ComparisonResult{
+			Status:     false,
+			Name:       "Compare actual event with expected event",
+			Explain:    fmt.Sprintf("Expected %d event(s), got %d instead.", len(tcs.ExpectedEvents), len(events)),
+			Path:       filepath.Base(tcs.File),
+			EventIndex: 0,
+		}
+		dataObserver := logstash_observer.NewDataObserver(comparisonResult, false, false)
+		liveProducer.Update(dataObserver)
+		return false, nil
+	}
+
+	// Check if test consit to validate that all event are dropped and so not failed before
+	if len(events) == 0 {
+		comparisonResult := &logstash_observer.ComparisonResult{
+			Status:     true,
+			Name:       "Compare actual event with expected event",
+			Explain:    "Drop all events",
+			Path:       filepath.Base(tcs.File),
+			EventIndex: 0,
+		}
+		dataObserver := logstash_observer.NewDataObserver(comparisonResult, false, false)
+		liveProducer.Update(dataObserver)
+		return true, nil
 	}
```

Comment:
Hum, we have few tests where we check that Logstash drop events because of is kind of noise.
Without this purpose, we can't explicity look that working (we look only result test case when it faield)